src/meta-srv/src/service/ddl.rs

```diff
@@ -185,3 +197,58 @@ async fn handle_create_table_route(
 
     router::TableRoute::try_from_raw(&peers, table_route).context(error::TableRouteConversionSnafu)
 }
+
+async fn handle_alter_table_task(
+    cluster_id: u64,
+    alter_table_task: AlterTableTask,
+    kv_store: KvStoreRef,
+    ddl_manager: DdlManagerRef,
+) -> Result<SubmitDdlTaskResponse> {
+    let table_id = alter_table_task
+        .alter_table
+        .table_id
+        .clone()
+        .context(error::UnexpectedSnafu {
+            violated: "expected table id ",
+        })?
+        .id;
+
+    let alter_table_request = alter_expr_to_request(table_id, alter_table_task.alter_table.clone())
+        .context(error::ConvertGrpcExprSnafu)?;
+
+    let (table_global_value, table_route_value) =
+        fetch_table(&kv_store, alter_table_request.table_ref())
+            .await?
+            .with_context(|| error::TableNotFoundSnafu {
+                name: alter_table_request.table_ref().to_string(),
+            })?;
+
+    let table_route = router::TableRoute::try_from_raw(
+        &table_route_value.peers,
+        table_route_value
+            .table_route
+            .context(error::UnexpectedSnafu {
+                violated: "expected table_route",
+            })?,
+    )
+    .context(error::TableRouteConversionSnafu)?;
+
+    let table_info = table_global_value.table_info;
+
+    let id = ddl_manager
+        .submit_alter_table_task(
+            cluster_id,
+            alter_table_task,
+            alter_table_request,
+            table_route,
+            table_info,
+        )
+        .await?;
+
+    info!("Table: {table_id} created via procedure_id {id:?}");
+
+    Ok(SubmitDdlTaskResponse {
```

Comment:
Need to watch the precedure done (or failed)?

Reply:
The inner function will watch the procedure.